gnomad/resources/resource_utils.py

```diff
@@ -267,46 +267,54 @@ def import_resource(self, overwrite: bool = True, **kwargs) -> None:
         )
 
 
-class BaseVersionedResource(BaseResource, ABC):
+class BaseVersionedResource:
     """
-    Abstract class for a versioned resource
+    Class for a versioned resource
 
-    The `path`/`import_args` attributes of the versioned resource are those of the default version of the resource.
+    The attributes and methods of the versioned resource are those of the default version of the resource.
     In addition, all versions of the resource are stored in the `versions` attribute.
 
-    :param default_version: The default version of this resource (must to be in the `versions` dict)
+    :param default_version: The default version of this resource (must be in the `versions` dict)
     :param versions: A dict of version name -> resource.
     """
 
+    resource_class = BaseResource
+
+    __slots__ = ["default_version", "versions"]
+
     def __init__(self, default_version: str, versions: Dict[str, BaseResource]):
-        if type(self) is BaseVersionedResource:
-            raise TypeError("Can't instantiate abstract class BaseVersionedResource")
+        default_resource = versions[default_version]
 
-        if default_version not in versions:
-            raise KeyError(
-                f"default_version {default_version} not found in versions dictionary passed to {self.__class__.__name__}."
-            )
+        for version_resource in versions.values():
+            if not isinstance(version_resource, self.resource_class):
+                raise TypeError(
+                    f"{self.__class.__name__} requires all versions to be of type {self.resource_class.__name__}"
+                )
 
-        for version_name, version_resource in versions.items():
-            if version_resource.__class__ not in self.__class__.__bases__:
+            if version_resource.__class__ is not default_resource.__class__:
                 raise TypeError(
-                    f"Cannot create a {self.__class__.__name__} resource with version {version_name} of type {version_resource.__class__.__name__}"
+                    f"{self.__class.__name__} requires all versions to be of the same type"
```

Comment:
Does it make sense to print what the `default_resource.__class__` is?

Reply:
I think I left that out because the "correct" resource type isn't necessarily that of the default version. It could be that the default version's type is wrong and the others are correct.